QueueUsingStacks.java

```diff
@@ -0,0 +1,78 @@
+class MyQueue {
+
+    Stack<Integer> s1;
+    Stack<Integer> s2;
+    /** Initialize your data structure here. */
+    public MyQueue() {
+        s1 = new Stack<Integer>();
+        s2 = new Stack<Integer>();
+
+    }
+
+    /** Push element x to the back of queue. */
+    public void push(int x) {
+
+        // Move all elements from s1 to s2
+        while (!s1.isEmpty())
+        {
+            s2.push(s1.pop());
+            //s1.pop();
+        }
+
+        // Push item into s1
+        s1.push(x);
+
+        // Push everything back to s1
+        while (!s2.isEmpty())
+        {
+            s1.push(s2.pop());
+            //s2.pop();
+        }
+    }
+
```

Comment:
Thats a good solution but you need to try both the solutions.
Now go for optimal one.